### PR TITLE
EC/ROCM: add support for using rocm memory from tl/upc

### PR DESCRIPTION
**Pull request by edgargabriel**
## What
Add support for using TL/UCP for ROCm memory

## Why ?
This commit allows among other things to use the Open MPI /ucc component for buffer located on ROCm devices.

### CODE REVIEW

File: src/components/ec/rocm/ec_rocm_executor_interruptible.c [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                                           stream));
+        if (ucc_unlikely(status != UCC_OK)) {
+            ec_error(&ucc_ec_rocm.super, "failed to start memcpy op");
+            goto free_task;
+        }
+
+        break;
+    case UCC_EE_EXECUTOR_TASK_TYPE_REDUCE:
+        /* temp workaround to avoid code duplication*/
+        printf("executor operation is supported task_type %d", task_args->task_type);
```
Comment (vspetrov) [inline src/components/ec/rocm/ec_rocm_executor_interruptible.c:89]: remove printf

**Review by vspetrov — APPROVED** [review]

File: src/components/ec/rocm/ec_rocm.c [hunk 2]
```diff
@@ -317,11 +456,24 @@ ucc_status_t ucc_ec_rocm_event_test(void *event)
...
+            ROCM_FUNC(hipStreamDestroy(ucc_ec_rocm.exec_streams[i]));
+        }
+        ucc_ec_rocm.exec_streams_initialized = 0;
+    }
+
+    ucc_mpool_cleanup(&ucc_ec_rocm.events, 1);
+    ucc_mpool_cleanup(&ucc_ec_rocm.strm_reqs, 1);
+    ucc_mpool_cleanup(&ucc_ec_rocm.executors, 1);
+    ucc_free(ucc_ec_rocm.exec_streams);
+    //ucc_spinlock_destroy(&ucc_ec_rocm.init_spinlock);
```
[the remark this answers is not in the document]
Comment (edgargabriel) [reply src/components/ec/rocm/ec_rocm.c]: done

File: src/components/ec/rocm/ec_rocm_executor_interruptible.c (hunk 1 above)
Comment (edgargabriel) [reply src/components/ec/rocm/ec_rocm_executor_interruptible.c]: done

**Review by Sergei-Lebedev — APPROVED** [review]

**Comment by vspetrov** [comment]
bot:retest

**Pull request merged**